Running a single test got harder

The changes introduced in #30975 unfortunately mean that it's harder to run a single test from the command line, and in particular the command that `REPRODUCE WITH` says will reproduce a failure will not work if the failing test was from the `:server:test` task. For instance:

```
$ ./gradlew :server:test -Dtests.class=org.elasticsearch.common.rounding.TimeZoneRoundingTests -Dtests.method="testUTCTimeUnitRounding"
...
> Task :server:testScriptDocValuesMissingV6Behaviour FAILED
   [junit4] <JUnit4> says jolly good day! Master seed: AA38CDFBB89D3357
==> Test Summary: 0 suites, 0 tests
   [junit4] Tests summary: 0 suites, 0 tests

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':server:testScriptDocValuesMissingV6Behaviour'.
> There were no executed tests: 0 suites, 0 tests
...
```

As a workaround, one can add `-x :server:testScriptDocValuesMissingV6Behaviour` to the command line.

--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -764,7 +764,7 @@ class BuildPlugin implements Plugin<Project> {
             additionalTest.configure(commonTestConfig(project))
             additionalTest.configure(config)
             additionalTest.dependsOn(project.tasks.testClasses)
-            test.dependsOn(additionalTest)
+            project.check.dependsOn(additionalTest)
         });
         return test
     }

--- a/server/build.gradle
+++ b/server/build.gradle
@@ -156,6 +156,16 @@ if (isEclipse) {
 compileJava.options.compilerArgs << "-Xlint:-cast,-deprecation,-rawtypes,-try,-unchecked"
 compileTestJava.options.compilerArgs << "-Xlint:-cast,-deprecation,-rawtypes,-try,-unchecked"
 
+// TODO: remove ScriptDocValuesMissingV6BehaviourTests in 7.0
+additionalTest('testScriptDocValuesMissingV6Behaviour'){
+  include '**/ScriptDocValuesMissingV6BehaviourTests.class'
+  systemProperty 'es.scripting.exception_for_missing_value', 'false'
+}
+test {
+  // these are tested explicitly in separate test tasks
+  exclude '**/*ScriptDocValuesMissingV6BehaviourTests.class'
+}
+
 forbiddenPatterns {
   exclude '**/*.json'
   exclude '**/*.jmx'
@@ -329,7 +339,7 @@ if (isEclipse == false || project.path == ":server-tests") {
   task integTest(type: RandomizedTestingTask,
                  group: JavaBasePlugin.VERIFICATION_GROUP,
                  description: 'Multi-node tests',
-                 dependsOn: test.dependsOn.collect()) {
+                 dependsOn: test.dependsOn) {
     configure(BuildPlugin.commonTestConfig(project))
     classpath = project.test.classpath
     testClassesDirs = project.test.testClassesDirs
@@ -339,12 +349,3 @@ if (isEclipse == false || project.path == ":server-tests") {
   integTest.mustRunAfter test
 }
 
-// TODO: remove ScriptDocValuesMissingV6BehaviourTests in 7.0
-additionalTest('testScriptDocValuesMissingV6Behaviour'){
-  include '**/ScriptDocValuesMissingV6BehaviourTests.class'
-  systemProperty 'es.scripting.exception_for_missing_value', 'false'
-}
-test {
-  // these are tested explicitly in separate test tasks
-  exclude '**/*ScriptDocValuesMissingV6BehaviourTests.class'
-}

